internal/receiver/discoveryreceiver/config.go

```diff
@@ -0,0 +1,162 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discoveryreceiver
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+	"go.uber.org/multierr"
+)
+
+var (
+	_ config.Receiver = (*Config)(nil)
+
+	allowedStatusTypes = []string{"successful", "partial", "failed"}
+	allowedStatuses    = func() map[string]struct{} {
+		sm := map[string]struct{}{}
+		for _, status := range allowedStatusTypes {
+			sm[status] = struct{}{}
+		}
+		return sm
+	}()
+
+	allowedMatchTypes = []string{"regexp", "strict", "expr"}
+)
+
+type Config struct {
+	// Receivers is a mapping of all receiver types to discovery and whose metrics
+	// and application statements to evaluate for assessing component status.
+	Receivers               map[config.ComponentID]ReceiverEntry `mapstructure:"receivers"`
```

Comment:
I don't understand this explanation. "Whose metrics" are those? Receivers? If so, I'd go with:
"Receivers is a mapping of all receiver types. Receiver metrics and application statements are used to evaluate and assess component status."

Reply:
I meant "receiver types to discover" but clarified a bit.

Reply:
I'd still break it in two sentences: 
Receivers is a mapping of all receivers to discover. Receiver metrics and application statements are evaluated to assess component status.

Reply:
The mapping is from the receivers (their names) to their evaluated metrics and statements (the `ReceiverEntry` and that contains `Status`).